lua/entities/gmod_wire_expression2/base/compiler.lua

```diff
@@ -1335,14 +1335,15 @@ function Compiler:InstrEVENT(args)
 		self:Error("Event '" .. name .. "' does not take arguments (" .. table.concat(extra_arg_types, ", ") .. ")", args)
 	end
 
-	for k, typeid in ipairs(event.args) do
+	for k, arg in ipairs(event.args) do
 		if not hargs[k] then
 			-- TODO: Maybe this should be a warning so that events can have extra params added without breaking old code?
-			self:Error("Event '" .. name .. "' missing argument #" .. k .. " of type " .. tps_pretty(typeid), args)
+			self:Error("Event '" .. name .. "' missing argument #" .. k .. " of type " .. tps_pretty(arg["type"]), args)
```

Comment:
You should be using `table.xyz` syntax rather than `table["xyz"]` which is unnecessary